projects/plugins/boost/app/modules/optimizations/minify/Minify.php

```diff
@@ -2,16 +2,13 @@
 
 namespace Automattic\Jetpack_Boost\Modules\Optimizations\Minify;
 
-use Automattic\Jetpack_Boost\Contracts\Pluggable;
+class Minify {
+	private static $called = false; // This class is called multiple times, so we need to make sure we only run some things once.
```

Comment:
This is only ever used inside `setup()`, so it has no reason to be a class static. Better to make it a local static variable in the function where it's used.

In fact, for readability I would suggest creating a method called `setup_once()` which has a static local variable in it to track whether it's been called or not - which can return early if it's already been called.

Then `setup` can include the stuff that needs to be setup per instance, and call `setup_once` to handle the special-case stuff.

Reply:
I went with the function approach you suggested. You can find it here:
https://github.com/Automattic/jetpack/blob/286067045c360986184d090cb638bc9956fd7bcf/projects/plugins/boost/app/lib/minify/functions-helpers.php#L267